bluesky/bundlers.py

```diff
@@ -408,39 +415,60 @@ async def save(self, msg):
                 hints=hints,
                 object_keys=object_keys,
             )
-            await self.emit(DocumentNames.descriptor, doc)
-            doc_logger.debug("[descriptor] document is emitted with name %r containing "
-                             "data keys %r (run_uid=%r)", name, data_keys.keys(),
-                             self._run_start_uid,
-                             extra={'doc_name': 'descriptor',
-                                    'run_uid': self._run_start_uid,
-                                    'data_keys': data_keys.keys()})
-            self._descriptors[desc_key] = (objs_read, doc)
+            await self.emit(DocumentNames.descriptor, descriptor_doc)
+            doc_logger.debug(
+                "[descriptor] document emitted with name %r containing "
+                "data keys %r (run_uid=%r)",
+                obj_name,
+                data_keys.keys(),
+                self._run_start_uid,
+                extra={
+                    'doc_name': 'descriptor',
+                    'run_uid': self._run_start_uid,
+                    'data_keys': data_keys.keys()}
+            )
+            self._descriptors[desc_key] = (objs_read, descriptor_doc)
 
-        descriptor_uid = doc["uid"]
+        descriptor_uid = descriptor_doc["uid"]
 
         # Resource and Datum documents
-        for name, doc in self._asset_docs_cache:
-            # Add a 'run_start' field to the resource document on its way out.
-            if name == "resource":
-                doc["run_start"] = self._run_start_uid
-            await self.emit(DocumentNames(name), doc)
+        for resource_or_datum_name, resource_or_datum_doc in self._asset_docs_cache:
+            # Add a 'run_start' field to resource documents on their way out
+            # since this field could not have been set correctly before this point.
+            if resource_or_datum_name == "resource":
+                resource_or_datum_doc["run_start"] = self._run_start_uid
+
+            doc_logger.debug(
+                "[%s] document emitted %r",
+                resource_or_datum_name,
+                resource_or_datum_doc,
+                extra={
+                    "doc_name": resource_or_datum_name,
+                    "run_uid": self._run_start_uid,
+                    "doc": resource_or_datum_doc
```

Comment:
When we last revamped logging here, there was lengthy discussion about how much to include. Considerations are readability of logs and also performance. (Making a large dict into a string for writing can impose noticeable overhead.) We decided to include the document name, the current Run uid, and any particular info especially likely to be useful for debugging, such as the data keys in an event or descriptor. In this case, I think we are justified in emitting the entire document for Resource and Datum because:

1. These document types never grow very large.
2. All of the information in the document is likely useful for debugging. No field stands out as less important than the others in this context.